runbotics-orchestrator/src/main/java/com/runbotics/service/ProcessInstanceQueryService.java

```diff
@@ -100,6 +102,30 @@ public Page<ProcessInstanceDTO> findByCriteria(ProcessInstanceCriteria criteria,
         return processInstanceRepository.findAll(specification, page).map(processInstanceMapper::toDto);
     }
 
+    /**
+     * Return a {@link Page} of {@link ProcessInstanceDTO} which matches the criteria from the database and also contains enity with provided id.
+     * @param id The id of entity which should appear in the {@link Page} of {@link ProcessInstanceDTO}
+     * @param criteria The object which holds all the filters, which the entities should match.
+     * @param page The page, which should be returned.
+     * @return the matching entities.
+     */
+    @Transactional(readOnly = true)
+    public Page<ProcessInstanceDTO> findByCriteriaWithSpecificInstance(UUID id, ProcessInstanceCriteria criteria, Pageable page) {
+        final Specification<ProcessInstance> specification = createSpecification(criteria);
+        List<ProcessInstance> myList = processInstanceRepository.findAll(specification, page.getSort());
+        int index = -1;
+        for (int i = 0; i < myList.size(); i++) {
+            if (myList.get(i).getId().equals(id)) {
+                index = i;
+                break;
+            }
+        }
+        int pageNumber = index < 0 ? 0 : index / page.getPageSize();
```

Comment:
<img width="914" alt="image" src="https://github.com/runbotics/runbotics/assets/69820069/33a20e32-463a-4f7f-b93d-9fcb370f4adc">

Reply:
It's probably page index but on the frontend it's treated as a page number because there's the addition of 1, if we want to make it consistent we should rework rest of them